### ISSUE TITLE
Add Support for Gyro/Motion Controls Using a Controller

### ISSUE THREAD

**Issue by moomeow96** [Lime3DS/Lime-3DS-Emulator#476]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What feature are you suggesting?

I'm requesting a feature that allows the user of the emulator to control the gyro sensor with the motion controls of a compatible controller (specifically for Android, but Windows too, if possible). 

### Why would this feature be useful?

Since most people use a controller for the emulator, they'll have an easier time using the gyro controls on the emulator, rather than having to grab the phone in order to use gyro controls, or making a keybind for them (eww).

**Comment by Emerald-Fox** [comment]
PC does already have Gyro support via CemuhookUDP or SDL
![image](https://github.com/user-attachments/assets/53809565-18fc-46de-997e-43c660aec349)

**Comment by moomeow96** [comment]
I know, but I'm looking for support added to Android. Is this possible?

**Comment by moomeow96** [comment]
I mean, Dolphin emulator did it recently, so I don't see why you can't.

**Comment by moomeow96** [comment]
Now I see why you can't lol

**Comment by JazzJohannes** [comment]
I second that, l would love to see gyro controls at least the tilt motion for racers

**Comment by moomeow96** [comment]
@JazzJohannes Yes, and it would especially make sense on Android because of the accelerometer found in most Android (and all iOS) phones.

**Comment by dm699** [comment]
> Now I see why you can't lol

Why is that?

**Comment by DavidRGriswold** [comment]
Android already automatically uses the built in gyroscope on phones for tilt controls. But adding it for external controllers specifically is a fair request. If that is what this issue is really requesting could you please adjust the language of the issue to fit that?

**Comment by moomeow96** [comment]
@dm699 Because Lime3DS is now Azahar

**Comment by Wildtick** [comment]
I would also like for this to be added for Android

**Comment by chadbanks** [comment]
I would also like to be able to use motion controls with my wireless controller on Android.

**Comment by moomeow96** [comment]
After some research, I have discovered that Android does not have native support for extrenal gyro, but it might be possible with a dongle that allows for gyro. If that exists, maybe we could work off that.

**Comment by DavidRGriswold** [comment]
This does not seem to be true, there is an API for accessing controller acceleromoters, at least on android 12 and higher. May not work on older devices. https://developer.android.com/games/sdk/game-controller/controller-features#motion-sensors I will put this on my list of things to look at in the future, but it will not be super high-priority for now. If somebody else wants to look into it, that should be the start!

**Comment by moomeow96** [comment]
Ok, sounds good!

**Comment by phly95** [comment]
A lot of controllers act as a Nintendo Switch controller (8bitdo, Shanwan, etc.) which should expose gyro to apps, right? Some people might be using tablets, have a phone mounted to something, or have an Android device connected to a TV, all of which would benefit from using the gyro from the controller itself.

I just ordered the Shanwan Q36 Pro expecting it to be the perfect pocket-sized controller for mounting to the seat in front of me in a car and such, and just found out that I can't use gyro that way sadly. It otherwise seems like a great pocket controller for 3DS games given that it's basically an 8bitdo Zero 2 with gyro and an analog stick where the Dpad would be. But I guess I'll have to wait on the gyro part for 3DS games.

On the bright side, the non-gyro telescopic controller I have is doing pretty good using the phone gyro despite it also spoofing a switch controller.
